llvm/test/SYCLPropagateAspectUsage/aggregate_types.ll

```diff
@@ -0,0 +1,71 @@
+; RUN: opt --PropagateAspectUsage < %s -S
+; | FileCheck %s
+;
+; Test checks that no metadata propagates because MyStruct
+; isn't used inside functions.
+
+; Optional
+%A = type { i32 }
+
+; Not optional
+%B = type { i32 }
+
+; Not optional
+%C = type { i32 }
+
+; Contains optional - %A
+%D1 = type { %A, %B, %C }
+
+; Doesn't contain optionals
+%D2 = type { %B, %C }
+
+; Contains %D which contains optional %A
+%E = type { %B, %C, %D1 }
+
+; Contains a pointer on %D which contains optional %A
+%F = type { %B, %C*, %D1* }
+
+; CHECK: dso_local spir_kernel void @kernel() !intel_used_aspects !1 {
+define dso_local spir_kernel void @kernelD1() {
+  call spir_func void @funcD1()
+  ret void
+}
+
+define dso_local spir_func void @funcD1() {
+  %tmp = alloca %D1
+  ret void
+}
+
+define dso_local spir_kernel void @kernelD2() {
```

Comment:
```suggestion
; CHECK: dso_local spir_kernel void @kernelD2() {
define dso_local spir_kernel void @kernelD2() {
```

To check that no metadata was attached to that kernel. Since the pass does propagation of aspects through the whole call stack, I suggest we append a `CHECK` line to _each_ function to check that each of them was handled correctly